Plugins/DotNET/NWN/Internal/Internal.cs

```diff
@@ -64,25 +65,25 @@ public static void OnClosure(ulong eid, uint oidSelf)
             Closures.Remove(eid);
```

Comment:
It would be nice to clear it back to OBJECT_INVALID, so we catch these in the future.

Reply:
Something like `finally { OBJECT_SELF = OBJECT_INVALID; }`?

Reply:
`finally` is generally useless if the `catch` branch doesn't re-throw. I'd do it as:

```csharp
uint old = OBJECT_SELF;
OBJECT_SELF = oidSelf;
try {...} catch(...) {...}
Closures.Remove(eid);
OBJECT_SELF = old;
```